microdata_validator/metadata_model.py

```diff
@@ -0,0 +1,349 @@
+            )
+        if self.name != other.name:
+            raise PatchingError(
+                'Can not change keyType name from '
+                f'"{self.name}" to "{other.name}"'
+            )
+        self.label = other.label
+        self.descritpion = other.description
+
+
+class CodeListItem:
+    category: str
+    code: str
+
+    def __init__(self, category: str, code: str):
+        self.category = category
+        self.code = code
+
+    def to_dict(self) -> dict:
+        return {"category": self.category, "code": self.code}
+
+    def patch(self, other: 'CodeListItem'):
+        if other is None:
+            raise PatchingError(
+                'Can not delete CodeListItem'
+            )
+        if self.code != other.code:
+            raise PatchingError(
+                'Can not change CodeListItem code from '
+                f'"{self.code}" to "{other.code}"'
+            )
+        self.category == other.category
+
+
+class ValueDomain:
+    description: str
+    unit_of_measure: str
+    code_list: List[CodeListItem]
+    missing_values: str
+
+    def __init__(self, value_domain_dict: dict):
+        self.description = value_domain_dict.get("description", None)
+        self.unit_of_measure = value_domain_dict.get("unitOfMeasure", None)
+
+        self.code_list = []
+        if "codeList" in value_domain_dict:
+            self.code_list = [
+                CodeListItem(item["category"], item["code"])
+                for item in value_domain_dict["codeList"]
+            ]
+
+        self.missing_values = []
+        if "missingValues" in value_domain_dict:
+            self.missing_values = [
+                missing_value for missing_value
+                in value_domain_dict["missingValues"]
+            ]
+
+    def to_dict(self) -> dict:
+        dict_representation = {
+            "description": self.description,
+            "unitOfMeasure": self.unit_of_measure,
+            "codeList": [code_item.to_dict() for code_item in self.code_list],
+            "missingValues": self.missing_values
+        }
+        return {
+            key: value for key, value in dict_representation.items()
+            if value not in [[], None]
+        }
+
+    def patch(self, other: 'ValueDomain'):
+        if other is None:
+            raise PatchingError(
+                'Can not delete ValueDomain'
+            )
+        if self.unit_of_measure != other.unit_of_measure:
+            raise PatchingError(
+                'Can not change ValueDomain unitOfMeasure from '
+                f'"{self.unit_of_measure}" to "{other.unit_of_measure}"'
+            )
+        if self.missing_values != other.missing_values:
+            raise PatchingError(
+                'Can not change ValueDomain missingValues from '
+                f'"{self.missing_values}" to "{other.missing_values}"'
+            )
+        if self.description is not None:
+            self.description == other.description
+        else:
```

Comment:
The code under else should not be a part of this if-else conditional.

Reply:
I'm pretty sure it should. If a ValueDomain is "described" it should contain a description and Optional(unitOfMeasure) or a codeList+missingValues

This has been named a "described value domain" and "enumerated value domain" in the other components. ([Line 18 here](https://github.com/statisticsnorway/microdata-dataset-builder/blob/master/dataset_builder/steps/dataset_transformer.py))

That being said, This check should be based on if a code list is present or not rather than a description, so I will update 👍🏻

Reply:
Oh, I see. Yes, the check should be more explisit based on code list existence 👍